sdk/cosmosdb/cosmos/samples-dev/ChangeFeedIterator/ChangeFeedIteratorEntireContainer.ts

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * @summary Demonstrates using a ChangeFeed for entire container
+ */
+
+import * as dotenv from "dotenv";
+dotenv.config();
+
+import { finish, handleError, logSampleHeader } from "../Shared/handleError";
+import {
+  CosmosClient,
+  PartitionKeyDefinitionVersion,
+  Container,
+  StatusCodes,
+  ChangeFeedIteratorOptions,
+  ChangeFeedResourceType,
+  ChangeFeedStartFrom,
+} from "@azure/cosmos";
+
+const key = process.env.COSMOS_KEY || "<cosmos key>";
+const endpoint = process.env.COSMOS_ENDPOINT || "<cosmos endpoint>";
+const databaseId = process.env.COSMOS_DATABASE || "<cosmos database>";
+const containerId = process.env.COSMOS_CONTAINER || "<cosmos container>";
+
+logSampleHeader("Change Feed");
+
+async function ingestData(container: Container, initialize: number, end: number) {
+  console.log("beginning data ingestion");
+  for (let i = initialize; i < end; i++) {
+    await container.items.create({ name: "sample1", key: i });
+    await container.items.create({ name: "sample2", key: i });
+    await container.items.create({ name: "sample3", key: i });
+    await container.items.create({ name: "sample4", key: i });
+  }
+  console.log("ingested items");
+}
+
+async function waitFor(milliseconds: number): Promise<void> {
+  return new Promise((resolve) => setTimeout(resolve, milliseconds));
+}
+
+// Establish a new instance of the CosmosClient to be used throughout this demo
+const client = new CosmosClient({ endpoint, key });
+
+async function iterateChangeFeedTillNow(container: Container): Promise<string> {
+  console.log("fetching changefeed until now");
+
+  const changeFeedIteratorOptions: ChangeFeedIteratorOptions = {
+    maxItemCount: 1,
+    changeFeedStartType: { startFrom: ChangeFeedStartFrom.Beginning },
+    changeFeedResource: { resource: ChangeFeedResourceType.Container },
+  };
+  const feedIterator = await container.items.getChangeFeedIterator(changeFeedIteratorOptions);
+
+  let continuationToken: string = "";
+
+  while (feedIterator.hasMoreResults) {
+    // infinite loop to check for new results. hasMoreResults is always true.
+    try {
+      const result = await feedIterator.ReadNextAsync();
+      continuationToken = result.continuationToken;
```

Comment:
nit: To avoid materialization of the continuation on each iteration, should we do this only on the NotModified case?

Reply:
Will update the sample.